fiftyone/core/collections.py

```diff
@@ -10548,42 +10523,132 @@ def create_index(self, field_or_spec, unique=False, wait=True, **kwargs):
+                replace_existing = True
+            elif _existing_unique or (unique == _existing_unique):
+                # Satisfactory index already exists
+                return _existing_name
+
+            # Handle default indexes
+            if (
+                _index_name in self._get_default_indexes(frames=is_frame_index)
+                and index_name != "filepath"  # allow 'filepath' to be modified
+            ):
+                raise ValueError(f"Cannot modify default index '{index_name}'")
+
+        # Convert existing index to unique, if necessary
+        if convert_to_unique:
+            logger.info(
+                f"Converting existing index '{index_name}' to unique "
+                f"on dataset '{self._dataset.name}'"
             )
 
-        name = coll.create_index(index_spec, unique=unique, **kwargs)
+            db = foo.get_db_conn()
+
+            try:
+                # https://www.mongodb.com/docs/manual/core/index-unique/convert-to-unique
+                db.command(
+                    "collMod",
+                    coll_name,
+                    index={"name": _existing_db_name, "prepareUnique": True},
+                )
+                db.command(
+                    "collMod",
+                    coll_name,
+                    index={"name": _existing_db_name, "unique": True},
+                )
+
+                return _existing_name
+            except:
+                if foo.get_db_version() < Version("6"):
+                    # index conversion was introduced in MongoDB 6
+                    replace_existing = True
+                else:
+                    db.command(
+                        "collMod",
+                        coll_name,
+                        index={
+                            "name": _existing_db_name,
+                            "prepareUnique": False,
+                        },
+                    )
+
+                    raise
+
+        # Drop existing index, if necessary
+        if replace_existing:
+            if not convert_to_unique:
+                logger.info(
+                    f"Replacing existing index '{index_name}' "
+                    f"on dataset '{self._dataset.name}'"
+                )
+
+            coll.drop_index(_existing_db_name)
+
+        # Create new index, if necessary
+        if not wait:
+            # Setting `w=0` sets `acknowledged=False` in pymongo
+            if is_frame_index:
+                coll = self._dataset._get_frame_collection(
+                    write_concern=WriteConcern(w=0)
+                )
+            else:
+                coll = self._dataset._get_sample_collection(
+                    write_concern=WriteConcern(w=0)
+                )
 
-        if single_field_index:
-            name = input_spec[0][0]
-        elif is_frame_index:
-            name = self._FRAMES_PREFIX + name
+        coll.create_index(index_spec, unique=unique, **kwargs)
```

Comment:
can we add a log to indicate that an index is being created? It could help debug slowdowns.

Reply:
Without deep knowledge of the code, it would be really difficult to know from the "application layer" or callers that indexes are being built. Can we at least have a debug log? This would really help with triaging performance issues.